keras_cv/layers/preprocessing/base_image_augmentation_layer.py

```diff
@@ -210,8 +211,28 @@ def augment_keypoints(self, keypoints, transformation, **kwargs):
         """
         raise NotImplementedError()
 
+    def augment_segmentation_mask(self, segmentation_mask, transformation, **kwargs):
+        """Augment a single image's segmentation mask during training.
+
+        Args:
+          segmentation_mask: 3D segmentation mask input tensor to the layer.
```

Comment:
For the segmentation_mask, what's the rank we expect for it? Is it [H, W, 1] or just [H, W]?

Reply:
The rank is variable based on the segmentation task. E.g. for binary segmentation this would be 2 (assuming one-hot encoding), and for multi-class segmentation this would be `classes`

Reply:
Using one-hot seems to be inconsistent with other values in the API, eg label shape in the test is [1] (and doesn't seems to be one-hot from the docstring). The class for the bbox is also attached as the 5th value in the list, and not using one-hot.

Reply:
The segmentation map doesn't *have* to be one-hot encoded. It could just as well be an integer class encoding, and the last dimension would be `1`.

Is there a reason we should force users to choose one or the other?

Reply:
we should probably start out segmentation map support with a literature review on the different segmentation map formats.  TFDS is a good first place to look, I believe there are some polygon based ones - it would be good to add some utilities for converting these as a starting point

Reply:
The two most popular datasets I'm seeing anecdotally in papers (as well as the top-2 on [papers with code](https://paperswithcode.com/datasets?task=semantic-segmentation)) are MS COCO and Cityscapes, which both use pixel-level segmaps.

FWIW, Cityscapes does also provide polygon-based segmentation information, but they call it "weakly annotated."

We could certainly add utilities for transforming common non-pixelmap formats to pixel maps, but it does seem like a pixelmap is the standard for segmentation masks, and these utilities may end up being dataset specific (and perhaps therefore belong in `keras_cv/datasets/`

Reply:
@qlzh727 I think the non-onehot approach is clearly reasonable for binary classification, but I think for multi-class segmentation it doesn't make sense.

But that's somewhat tangential here, because it shouldn't affect the API, as we should support any number of channels in the segmentation mask.

Reply:
For MS coco, I think they choose to use a RGB value to encode the mask, so that the mask can be saved just images (for easy distribution), but user will still have to decode the class ids by `ids=R+G*256+B*256^2`. The extra step of decode is kind of annoying and other datasets didn't follow that. 

Do you have an example dataset of multi-class segmentation mask? I was always assuming that the mask is single classed (not binary necessarily), but multi-class is doesn't work with [H, W, 1].

Reply:
It looks like this is how the [cityscapes](https://www.cityscapes-dataset.com/dataset-overview/#features) dataset is encoded, but I haven't confirmed that by building my own segmentation map for it.

the examples I've seen which use this dataset do use a [H, W, Classes] format for training (e.g. [this one](https://github.com/srihari-humbarwadi/cityscapes-segmentation-with-Unet))

Reply:
Btw, same for https://www.tensorflow.org/datasets/catalog/cityscapes, which has `'segmentation_label': Image(shape=(1024, 2048, 1), dtype=tf.uint8)`

Reply:
Also, if you check the latest implementation from the same author, the mask seems to be a 1-channel PNG image. https://github.com/srihari-humbarwadi/DeepLabV3_Plus-Tensorflow2.0/blob/f06df320aebe4aeeeccfa789340b14c0df163b09/train.py#L50

The [H, W, 1] is more aligned with user's intuition (for label), and [H, W, C] is bit unclear whether the data is one-hot, or actually multi-labeled. 

Having said that, I think we should document this in the docstring, that the segmentation map should be rank 3, and [H, W, 1] (for single class/labeled) or [H, W, C] in a very uncommon case.

Reply:
Btw, the one-hot encoding does have some implications for the KPLs, eg mixup, which will mix up the labels as well. The layer will perform quite differently if the mask is one-hot encoded.

Reply:
Sounds good -- thank you for the explanation! I'm updating the docstrings and will update the tests to use [H, W, 1]